lib/spack/spack/database.py

```diff
@@ -442,12 +467,18 @@ def _read_suppress_error():
                     tty.debug(
                         'RECONSTRUCTING FROM SPEC.YAML: {0}'.format(spec))
                     explicit = True
+                    inst_time = _now()
```

Comment:
Is the right thing to do really to update old installations that don't have an install time with the time of the last reindex?  Seems like that could led to surprising behavior.  What about the creation date of the package's install prefix?  That way people could just `spack reindex` and get reasonable install times, even for old stuff.